back/src/product/Product.Service.test.ts

```diff
@@ -0,0 +1,200 @@
+import "reflect-metadata";
+import { ProductService } from "./Product.Service";
+// import { createProductInputMock } from "./mocks/productInputMock";
+import {
+    createNewProductInputMock,
+    createTestCategoryInput,
+    createTestProductInput,
+} from "./mocks/productMock";
+import { testDbSetupt, testDbTeardown } from "../testUtils/testhelper";
+import { Product } from "./entity/Product";
+import { DataSource, Repository } from "typeorm";
+import { CategoryService } from "../category/Category.Service";
+import { Category } from "../category/entity/Category";
+import { UpdateProductInput } from "./inputs/UpdateProductInput";
+import { CreateProductInput } from "./inputs/CreateProductInput";
+
+jest.mock("../category/Category.Service");
+
+describe("ProductService", () => {
+    let productService: ProductService;
+    let mockCategoryService: jest.Mocked<CategoryService>;
+    let testDataSource: DataSource;
+    let categoryRepository: Repository<Category>;
+    let productRepository: Repository<Product>;
+    let testCategory: Category;
+    let testProduct: Product;
+    let createdProduct: Product;
+
+    beforeAll(async () => {
+        // Initialise the test database and the test repositories
+        testDataSource = testDbSetupt();
+
+        await testDataSource.initialize();
+
+        if (testDataSource.isInitialized) {
```

Comment:
J'ai un petit soucis avec ce if. Si il y a une erreur, il ne se passera rien car il n'y a pas de else. Après j'imagine que tu as mis ce if pour que TS soit content. Dans ce cas, la bonne pratique serait plutôt de faire un if dans le cas où il y a une erreur. Donc un truc dans ce style :

```suggestion
        if (!testDataSource.isInitialized) {
            // throw Error ou console.error() ou autre
        }
        // Reste du code
```
C'est un peu plus clean, car la tout ton code est dans un if